src/player_hardcoded_effects.cpp

```diff
@@ -1001,8 +1001,12 @@ void player::hardcoded_effects( effect &it )
         }
 
         // TODO: Move this to update_needs when NPCs can mutate
+        bool compatible_weather_types = g->weather == WEATHER_CLEAR || g->weather == WEATHER_SUNNY
```

Comment:
You can move variable declaration to inner scope of next if as it is only used there.

Reply:
Sorry, I've incorrectly read some lines. Everything is fine. Please ignore my comment.